src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8051,24 +8055,20 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
             }
 
             bool argIsIndex = argIsIndexNotRange.Value();
-            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
-            if (!TryFindIndexOrRangeIndexerPattern(receiverOpt, receiverType, argIsIndex: argIsIndex,
-                out PropertySymbol? lengthOrCountProperty, out Symbol? patternSymbol, diagnostics, ref useSiteInfo))
+            if (!TryFindIndexOrRangeIndexerFallback(syntax, receiverOpt, receiverType, argIsIndex: argIsIndex,
+                out PropertySymbol? lengthOrCountProperty, out Symbol? patternSymbol, diagnostics))
             {
                 return false;
             }
 
-            patternIndexerAccess = new BoundIndexOrRangePatternIndexerAccess(
+            indexerFallbackAccess = new BoundIndexOrRangeIndexerFallbackAccess(
                 syntax,
                 receiverOpt,
                 lengthOrCountProperty,
                 patternSymbol,
                 BindToNaturalType(argument, diagnostics),
                 patternSymbol.GetTypeOrReturnType().Type);
 
-            ReportDiagnosticsIfObsolete(diagnostics, patternSymbol, syntax, hasBaseReceiver: false);
```

Comment:
>ReportDiagnosticsIfObsolete(diagnostics, patternSymbol, syntax, hasBaseReceiver: false); [](http://example.com/codeflow?start=12&length=88)

Where do we perform these checks now?

Reply:
It feels like we should also perform checks normally done by CheckReceiverAndRuntimeSupportForSymbolAccess helper.

Reply:
> Where do we perform these checks now? (`ReportDiagnosticsIfObsolete(diagnostics, patternSymbol, syntax, hasBaseReceiver: false);`)

`TryBindLengthOrCount` does `ReportDiagnosticsIfObsolete(diagnostics, lengthOrCountProperty, syntax, hasBaseReceiver: false);`
and
`TryBindIndexOrRangeImplicitIndexer` reports via `BindIndexerOrIndexedPropertyAccess` for the `this[int]` case, and will report via call to bind method for `Slice` and `Substring` cases (tracked by another comment).

> It feels like we should also perform checks normally done by CheckReceiverAndRuntimeSupportForSymbolAccess helper.

`CheckReceiverAndRuntimeSupportForSymbolAccess` is called by `CheckPropertyValueKind`, so that should all be handled by `CheckValue` on the implicit indexer access.